frontend/src/components/modalContent/AddPartyModal.js

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getFormValues, submit, reset } from "redux-form";
+import { Steps, Button, Popconfirm } from "antd";
+import * as FORM from "@/constants/forms";
+import CustomPropTypes from "@/customPropTypes";
+import { createParty } from "@/actionCreators/partiesActionCreator";
+import AddFullPartyForm from "@/components/Forms/parties/AddFullPartyForm";
+
+const { Step } = Steps;
+
+const propTypes = {
+  fetchData: PropTypes.func.isRequired,
+  createParty: PropTypes.func.isRequired,
+  submit: PropTypes.func.isRequired,
+  reset: PropTypes.func.isRequired,
+  title: PropTypes.string,
+  addPartyFormValues: PropTypes.objectOf(PropTypes.strings),
+  addPartyForm: PropTypes.objectOf(CustomPropTypes.genericFormState),
+};
+
+const defaultProps = {
+  addPartyFormValues: {},
+  addPartyForm: {},
+  title: "",
+};
+
+export class AddPartyModal extends Component {
+  state = { isPerson: true, current: 0 };
+
+  togglePartyChange = (value) => {
+    this.setState({ isPerson: value.target.value });
+  };
+
+  handlePartySubmit = (event, addAnother) => {
+    event.preventDefault();
+    const type = this.state.isPerson ? "PER" : "ORG";
+    const payload = { type, ...this.props.addPartyFormValues };
+    return this.props
+      .createParty(payload)
+      .then(() => {
+        this.props.reset(FORM.ADD_FULL_PARTY);
+        if (addAnother) {
+          this.prev();
+        } else {
+          this.props.fetchData();
+          this.props.closeModal();
+        }
+      })
+      .catch(() => {
+        this.prev();
+      });
+  };
+
+  next() {
+    if (!this.props.addPartyForm.syncErrors) {
+      const current = this.state.current + 1;
+      this.setState({ current });
+    } else {
+      // submit form to trigger validation errors.... alternate suggestions??
+      this.props.submit(FORM.ADD_FULL_PARTY);
```

Comment:
If the validation errors are client-side then you should be able to update the validation errors to show up when the field is dirty (maybe with a 1-3 second delay?)

Reply:
Usually, yes. 
The way that redux-forms works, if it the form submission fails it shows all validations errors, but this form is bypassing the built-in features and submitting it from an outside component, instead of having the submit function connected to the form itself. (where all the built-in magic happens)

In this situation, I was trying to force the submission to fail to get all the form benefit
Redux-forms has a whole host of functions I could use, but none of them seemed to force the display of the errors, except trying to submit the invalid form and letting redux-forms do its thing. 
(https://redux-form.com/6.4.3/docs/api/reduxform.md/)

We can take this offline If I didn't explain it well enough!